python/sparkdl/transformers/tf_tensor.py

```diff
@@ -0,0 +1,140 @@
+        return self._set(**kwargs)
+
+
+    def _convertInternal(self):
+        assert self.isDefined(self.inputMapping) and self.isDefined(self.outputMapping), \
+            "inputMapping and outputMapping must be defined"
+
+        _maybe_graph = self.getTFGraph()
+        with IsolatedSession(graph=_maybe_graph) as issn:
+            if self.isDefined(self.exportDir):
+                assert _maybe_graph is None
+                assert not self.isDefined(self.tfCheckpointDir)
+                tag_set = self.getTagSet().split(',')
+                tf.saved_model.loader.load(issn.sess, tag_set, self.getExportDir())
+            elif self.isDefined(self.tfCheckpointDir):
+                assert _maybe_graph is None
+                ckpt_dir = self.getTFCheckpointDir()
+                ckpt_path = tf.train.latest_checkpoint(ckpt_dir)
+                print('using checkpoint path from {} as {}'.format(ckpt_dir, ckpt_path))
+                saver = tf.train.import_meta_graph("{}.meta".format(ckpt_path), clear_devices=True)
+                saver.restore(issn.sess, ckpt_path)
+            else:
+                assert _maybe_graph is not None
+
+            sig_def = None
+            if self.isDefined(self.signatureDefKey):
+                meta_graph_def = tf.train.export_meta_graph(issn.graph)
+                sig_def = tf.contrib.saved_model.get_signature_def_by_key(
+                    meta_graph_def, self.getSignatureDefKey())
+                assert sig_def is not None
+
+            feeds = []
+            for _, tnsr_or_sig in self.getInputMapping():
+                if sig_def:
+                    tnsr_name = sig_def.inputs[tnsr_or_sig]
+                else:
+                    tnsr_name = tfx.tensor_name(issn.graph, tnsr_or_sig)
+                feeds.append(tfx.get_tensor(issn.graph, tnsr_name))
+
+            fetches = []
+            self.output_renaming = {}
+            for tnsr, output_colname in self.getOutputMapping():
+                if sig_def:
+                    tnsr_name = tfx.tensor_name(issn.graph, tnsr)
+                    tnsr = sig_def.outputs[tnsr_name]
+                fetches.append(tfx.get_tensor(issn.graph, tnsr))
+                tf_expected_colname = tfx.op_name(issn.graph, tnsr)
+                self.output_renaming[tf_expected_colname] = output_colname
+
+            self.gfn = issn.asGraphFunction(feeds, fetches, strip_and_freeze=True)
+
+
+    def _transform(self, dataset):
+        self._convertInternal()
+
+        with IsolatedSession() as issn:
+            analyzed_df = tfs.analyze(dataset)
+            _, fetches = issn.importGraphFunction(self.gfn, prefix='')
+            feed_dict = dict([(tnsr_name, col_name) for col_name, tnsr_name in self.getInputMapping()])
+            out_df = tfs.map_blocks(fetches, analyzed_df, feed_dict=feed_dict)
+
+            # We still have to rename output columns
+            for old_colname, new_colname in self.output_renaming.items():
```

Comment:
We still need to rename output columns. But that is consistent with the `map_row` API.